packages/components/src/components/post-menu/post-menu.tsx

```diff
@@ -0,0 +1,194 @@
+import { Component, Element, Event, EventEmitter, h, Host, Method, Prop, State } from '@stencil/core';
+import { Placement } from '@floating-ui/dom';
+import { version } from '@root/package.json';
+
+/**
+ * @slot default - Slot for placing content inside the menu.
+ */
+
+let menuInstances = 0;
+
+@Component({
+  tag: 'post-menu',
+  styleUrl: 'post-menu.scss',
+  shadow: true,
+})
+export class PostMenu {
+  private popoverRef: HTMLPostPopovercontainerElement;
+
+  private readonly KEYCODES = {
+    SPACE: 'Space',
+    LEFT: 'ArrowLeft',
+    UP: 'ArrowUp',
+    RIGHT: 'ArrowRight',
+    DOWN: 'ArrowDown',
+    HOME: 'Home',
+    END: 'End'
+  };
+
+  @Element() host: HTMLPostMenuElement;
+
+  /**
+   * Defines the placement of the popover according to the floating-ui options available at https://floating-ui.com/docs/computePosition#placement.
+   */
+  @Prop() readonly placement?: Placement = 'bottom';
+
+  /**
+   * Tracks the visibility state of the menu (true if visible, false if hidden).
+   */
+  @State() isVisible: boolean = false;
+
+  /**
+   * Emits when the menu is shown or hidden.
+   * The emitted boolean value indicates the menu's visibility state.
+   */
+  @Event() toggleMenu: EventEmitter<boolean>;
+
+  connectedCallback() {
+    if (menuInstances === 0) {
+      this.host.addEventListener('keydown', this.handleKeyDown);
+    }
+
+    menuInstances++;
+  }
+
+  disconnectedCallback() {
+    menuInstances--;
+
+    if (menuInstances === 0) {
+      this.host.removeEventListener('keydown', this.handleKeyDown);
+    }
+  }
+
+  private handleKeyDown = (e: KeyboardEvent) => {
+    if (['ArrowDown', 'ArrowUp', 'ArrowLeft', 'ArrowRight', 'Space', 'Escape', 'Home', 'End'].includes(e.key)) {
+      this.controlKeyDownHandler(e);
+    }
+  };
+
+  private controlKeyDownHandler(e: KeyboardEvent) {
+    e.stopPropagation();
+  
+    const focusableItems = this.getSlottedItems();
+    if (!focusableItems.length) return;
+  
+    const currentFocusedElement = document.activeElement as HTMLElement;
+    let currentIndex = focusableItems.findIndex(el => el === currentFocusedElement);
+  
+    if (Object.values(this.KEYCODES).includes(e.code)) e.preventDefault();
+  
+    switch (e.code) {
+      case this.KEYCODES.UP:
+      case this.KEYCODES.LEFT:
+        if (currentIndex > 0) {
+          currentIndex = currentIndex - 1;
+        }
```

Comment:
You can use the modulo operator to wrap from the last item to the first. Here is an example: https://github.com/leamcg/autocomplete/blob/main/src/app/components/autocomplete/autocomplete.component.ts#L26